pybuildkite/client.py

```diff
@@ -72,7 +72,7 @@ def request(
         if with_pagination:
             response = self._get_paginated_response(response)
             return response
-        if method == "DELETE":
+        if method == "DELETE" or response.status_code == 204:
             return response.ok
         if headers.get("Accept") is None or headers.get("Accept") == "application/json":
```

Comment:
Could we add a check to ensure `response.headers.get('content-type')` is not `None` before trying to access `response.json()` to prevent this from happening again?